### PR TITLE
feat(gcpdisk-csi-driver): update images

### PR DESCRIPTION
**Pull request by sebbrandt87**
- logic for same image tag being used for snapshotter as
snapshot-controller
- updated images to compatible latest
- added replicas for SnapshotController
- only leader-election enabled if replicas > 1
- added CRD install for v1beta1
- added tolerations

Notes:
atm with this an "upgrade" would fail, and there would be the need to
manual remove the v1alpha1 snapshot CRDs. Which would also purge
snapshots. This must be documented.

changes as for #580 #581

[D2IQ-68177] #comment See GitHub

[D2IQ-68177]: https://jira.d2iq.com/browse/D2IQ-68177

### CODE REVIEW

File: stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml [hunk 1]
```diff
@@ -15,52 +15,48 @@ spec:
...
       priorityClassName: system-cluster-critical
+{{- if .Values.tolerations }}
+      tolerations:
+{{ toYaml .Values.tolerations | indent 8 }}
+{{- end }}
       containers:
         - name: csi-provisioner
           image: {{ .Values.provisioner.image.repository }}:{{ .Values.provisioner.image.tag }}
           args:
+            - --v=5
```
Comment (sreis) [inline stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml:30]: Configurable via `values.yaml`?

Comment (sreis) [reply stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml:30]: Just noticed it seems to be the same setting in the other csi drivers. Maybe it should be a setting available for all of them? wdyt?

File: stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml [hunk 2]
```diff
@@ -15,52 +15,48 @@ spec:
...
-            - --v=5
-          env:
-            - name: ADDRESS
-              value: /csi/csi.sock
-          volumeMounts:
-            - name: socket-dir
-              mountPath: /csi
         - name: csi-snapshotter
-          image: {{ .Values.snapshotter.image.repository }}:{{ .Values.snapshotter.image.tag }}
+          image: "{{ .Values.snapshotter.image.repository }}:{{ include "gcpdisk.snapshotter.image.tag" . }}"
```
Comment (hectorj2f) [inline stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml:52]: why did you change from include to `{{ .Values.snapshotter.image.tag }}` ?

File: stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml [hunk 3]
```diff
@@ -15,52 +15,48 @@ spec:
...
           args:
+            - --v=5
             - --csi-address=$(ADDRESS)
+            - --metrics-address=:22014
```
Comment (hectorj2f) [inline stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml:56]: are these ports the default ones ? why to set them then ?

File: stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml [hunk 4]
```diff
@@ -88,6 +88,10 @@ spec:
           args:
             - --v=5
             - --csi-address=$(ADDRESS)
+            - --metrics-address=:22013
```
Comment (hectorj2f) [inline stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml:91]: if these ports are the default ones, I don't see much interest on adding them

File: stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml (hunk 1 above)
Comment (sebbrandt87) [reply stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml:30]: could be done for all sure, but it is not right now.
We can tackle it in another ticket I will create.

File: stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml (hunk 3 above)
Comment (sebbrandt87) [reply stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml:56]: `--metrics-address: The TCP network address where the prometheus metrics endpoint will run (example: :8080 which corresponds to port 8080 on local host). The default is empty string, which means metrics endpoint is disabled.`

File: stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml (hunk 4 above)
Comment (sebbrandt87) [reply stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml:91]: see https://github.com/mesosphere/charts/pull/584/files#r430020207

File: stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml (hunk 1 above)
Comment (sebbrandt87) [reply stable/gcpdisk-csi-driver/templates/csi-gcpdisk-controller.yaml:30]: https://jira.d2iq.com/browse/D2IQ-68803
